### PR TITLE
Fix edge case where entities found for preview is empty

### PR DESCRIPTION
**Pull request by yizheliu-amazon**
*Issue #, if available:*

*Description of changes:*
Fix edge case where entities found for preview is empty.

Have verified on test domain that empty result is returned with this fix.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/AnomalyDetectorRunner.java [hunk 1]
```diff
@@ -72,6 +72,9 @@ public void executeDetector(AnomalyDetector detector, Instant startTime, Instant
         if (categoryField != null && !categoryField.isEmpty()) {
             featureManager.getPreviewEntities(detector, startTime.toEpochMilli(), endTime.toEpochMilli(), ActionListener.wrap(entities -> {
 
+                if (entities == null || entities.isEmpty()) {
+                    listener.onResponse(Collections.emptyList());
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/AnomalyDetectorRunner.java:76]: Should we return empty results or some failure like this https://github.com/opendistro-for-elasticsearch/anomaly-detection/blob/master/src/main/java/com/amazon/opendistroforelasticsearch/ad/feature/FeatureManager.java#L525

Comment (yizheliu-amazon) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/AnomalyDetectorRunner.java:76]: Such failure will be caught and empty result is still returned instead in [onFailure()](https://github.com/opendistro-for-elasticsearch/anomaly-detection/blob/master/src/main/java/com/amazon/opendistroforelasticsearch/ad/AnomalyDetectorRunner.java#L117-L119)

I think it is okay to just return empty result

**Review by ylwu-amzn — APPROVED** [review]

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/AnomalyDetectorRunner.java (hunk 1 above)
Comment (kaituo) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/AnomalyDetectorRunner.java:76]: Could you tell me why it would end up calling onFailure?  listener is the rest channel.

Comment (yizheliu-amazon) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/AnomalyDetectorRunner.java:76]: If entities is empty, code execution won't go into the for loop [below](https://github.com/opendistro-for-elasticsearch/anomaly-detection/blob/master/src/main/java/com/amazon/opendistroforelasticsearch/ad/AnomalyDetectorRunner.java#L87-L103), and `listener` has no chance to call `onFailure` or `onResponse`

Comment (kaituo) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/AnomalyDetectorRunner.java:76]: I feel it is better to return some failure and show that on kibana than return empty result.  Customers may wonder why it returns nothing.

Comment (yizheliu-amazon) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/AnomalyDetectorRunner.java:76]: if empty list is returned, existing Kibana will show corresponding message.

Comment (kaituo) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/AnomalyDetectorRunner.java:76]: Can you add a comment explaining why we want to return empty responses instead of errors?  It helps code readability.

**Review by kaituo — APPROVED** [review]
